pkg/controller/pod.go

```diff
@@ -274,6 +276,15 @@ func (pnc *PodNetworksController) addNetworks(dynamicAttachmentRequest *DynamicA
 		}
 		klog.Infof("response: %v", *response.Result)
 
+		newIfaceStatus, err := annotations.AddDynamicIfaceToStatus(pod, netToAdd, response)
+		if err != nil {
+			return fmt.Errorf("failed to compute the updated network status: %v", err)
+		}
+
+		if err := pnc.updatePodNetworkStatus(pod, newIfaceStatus); err != nil {
```

Comment:
We should reconstruct and retry on update conflict error so we are sure we take whatever others are updating.

https://pkg.go.dev/k8s.io/client-go/util/retry#RetryOnConflict

Reply:
If we fail, we re-queue the request on the controller queue, and the whole thing will be re-tried.